pkg/server/endpoints/authorized_entryfetcher.go

```diff
@@ -22,33 +24,44 @@ var _ api.AuthorizedEntryFetcher = (*AuthorizedEntryFetcherWithEventsBasedCache)
 const buildCachePageSize = 10000
 
 type AuthorizedEntryFetcherWithEventsBasedCache struct {
-	cache                        *authorizedentries.Cache
-	clk                          clock.Clock
-	log                          logrus.FieldLogger
-	ds                           datastore.DataStore
-	cacheReloadInterval          time.Duration
-	pruneEventsOlderThan         time.Duration
-	lastRegistrationEntryEventID uint
-	lastAttestedNodeEventID      uint
+	mu                                  sync.RWMutex
+	cache                               *authorizedentries.Cache
+	clk                                 clock.Clock
+	log                                 logrus.FieldLogger
+	ds                                  datastore.DataStore
+	cacheReloadInterval                 time.Duration
+	pruneEventsOlderThan                time.Duration
+	sqlTransactionTimeout               time.Duration
+	lastRegistrationEntryEventID        uint
+	lastAttestedNodeEventID             uint
+	missedRegistrationEntryEvents       map[uint]time.Time
+	missedAttestedNodeEvents            map[uint]time.Time
+	receivedFirstRegistrationEntryEvent bool
+	receivedFirstAttestedNodeEvent      bool
 }
 
-func NewAuthorizedEntryFetcherWithEventsBasedCache(ctx context.Context, log logrus.FieldLogger, clk clock.Clock, ds datastore.DataStore, cacheReloadInterval, pruneEventsOlderThan time.Duration) (*AuthorizedEntryFetcherWithEventsBasedCache, error) {
+func NewAuthorizedEntryFetcherWithEventsBasedCache(ctx context.Context, log logrus.FieldLogger, clk clock.Clock, ds datastore.DataStore, cacheReloadInterval, pruneEventsOlderThan, sqlTransactionTimeout time.Duration) (*AuthorizedEntryFetcherWithEventsBasedCache, error) {
 	log.Info("Building event-based in-memory entry cache")
-	cache, lastRegistrationEntryEventID, lastAttestedNodeEventID, err := buildCache(ctx, ds, clk)
+	cache, receivedFirstRegistrationEntryEvent, lastRegistrationEntryEventID, missedRegistrationEntryEvents, receivedFirstAttestedNodeEvent, lastAttestedNodeEventID, missedAttestedNodeEvents, err := buildCache(ctx, ds, clk)
```

Comment:
this is too much for a return
it is better to return a struct here, or reduce the scope of this buildCache